### PR TITLE
Bevy 0.15

### PR DESCRIPTION
**Pull request by bas-ie**
It's time for Bevy 0.15! Thanks so much for your work on `hexx`.

### CODE REVIEW

**Comment by richchurcher** [comment]
(Er, slightly premature PR lol... more coming!)

File: examples/3d_columns.rs [hunk 1]
```diff
@@ -28,11 +31,35 @@ pub fn main() {
...
+    fn from(material: StandardMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
+impl From<&StandardMaterialHandle> for AssetId<StandardMaterial> {
+    fn from(material: &StandardMaterialHandle) -> Self {
+        material.id()
+    }
+}
```
Comment (ManevilleF) [inline examples/3d_columns.rs:56]: I don't think this type, added to most 3d examples is needed. we can keep storing a `Handle<StandardMaterial>` and instantiate a `MeshMaterial3d`

File: examples/3d_picking.rs [hunk 2]
```diff
@@ -27,6 +30,30 @@ pub fn main() {
...
+        material.id()
+    }
+}
+
+impl From<&StandardMaterialHandle> for AssetId<StandardMaterial> {
+    fn from(material: &StandardMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
```
Comment (ManevilleF) [inline examples/3d_picking.rs:56]: Here again

File: examples/a_star.rs [hunk 3]
```diff
@@ -26,20 +29,44 @@ pub fn main() {
...
+        material.id()
+    }
+}
+
+impl From<&ColorMaterialHandle> for AssetId<ColorMaterial> {
+    fn from(material: &ColorMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
```
Comment (ManevilleF) [inline examples/a_star.rs:55]: Same here, with `MeshMaterial2d`

File: examples/field_of_view.rs [hunk 4]
```diff
@@ -26,20 +29,44 @@ pub fn main() {
...
+        material.id()
+    }
+}
+
+impl From<&ColorMaterialHandle> for AssetId<ColorMaterial> {
+    fn from(material: &ColorMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
```
Comment (ManevilleF) [inline examples/field_of_view.rs:55]: Again here

File: examples/hex_area.rs [hunk 5]
```diff
@@ -31,6 +34,30 @@ struct HexArea {
...
+        material.id()
+    }
+}
+
+impl From<&ColorMaterialHandle> for AssetId<ColorMaterial> {
+    fn from(material: &ColorMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
```
Comment (ManevilleF) [inline examples/hex_area.rs:60]: Again here

File: examples/hex_grid.rs [hunk 6]
```diff
@@ -39,22 +42,46 @@ struct HighlightedHexes {
...
+        material.id()
+    }
+}
+
+impl From<&ColorMaterialHandle> for AssetId<ColorMaterial> {
+    fn from(material: &ColorMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
```
Comment (ManevilleF) [inline examples/hex_grid.rs:68]: Again here

File: examples/shapes.rs [hunk 7]
```diff
@@ -31,11 +32,35 @@ fn main() {
...
+    }
+}
+
+impl From<&ColorMaterialHandle> for AssetId<ColorMaterial> {
+    fn from(material: &ColorMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
 #[derive(Resource)]
```
Comment (ManevilleF) [inline examples/shapes.rs:59]: Again here

File: examples/wrap_map.rs [hunk 8]
```diff
@@ -26,18 +29,42 @@ pub fn main() {
...
+        material.id()
+    }
+}
+
+impl From<&ColorMaterialHandle> for AssetId<ColorMaterial> {
+    fn from(material: &ColorMaterialHandle) -> Self {
+        material.id()
+    }
+}
+
```
Comment (ManevilleF) [inline examples/wrap_map.rs:55]: Again here

**Review by ManevilleF — CHANGES_REQUESTED** [review]
Thank you very much for this PR ! Seeing how many changed bevy 0.15 brought I really appreciate the time you spent updating the examples.
I have 2 comment:
- I don't think `bevy_reflect` is needed for all the examples
- The material wrappers you added in some examples are not needed either, we can use the `MaterialMesh3d/2d` components directly

As you already spent time on this, I can handle the changes so let me know if you'd rather fix those comments yourself

File: Cargo.toml [hunk 9]
```diff
@@ -81,10 +82,12 @@ unexpected_cfgs = { level = "warn", check-cfg = [
 [[example]]
 name = "hex_grid"
 path = "examples/hex_grid.rs"
+required-features = ["bevy_reflect"]
```
[the remark this answers is not in the document]
Comment (bas-ie) [reply Cargo.toml:85]: See below, enables wrapper components: but we could also go another way :slightly_smiling_face:

**Comment by bas-ie** [comment]
Yes, apologies, I was playing around with patterns now that `Handle<T>` is no longer a component. I'll rework this now.

**Comment by bas-ie** [comment]
There, that should be a bit better! Only thing we're still waiting on is a release for `bevy_inspector_egui`, after which I can update the examples that use it. I see they merged https://github.com/jakobhellermann/bevy-inspector-egui/commit/47f1cb6c5b63b47c40c9cbbd467422205319390e yesterday so they should be close!

**Comment by bas-ie** [comment]
I think that's the last of it? Let me know if I missed anything.

**Review by ManevilleF — APPROVED** [review]
There is an issue with doctests I'll fix in a followup, seems unrelated to your work
LGTM !

**Pull request merged**